catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/plugins/UpdateParentTest.java

```
@@ -0,0 +1,101 @@
+
+        UpdateParent.UpdateField updateField = mock(UpdateParent.UpdateField.class);
+
+        UpdateParent updateParent = new UpdateParent(updateField);
+
+        Context context = mock(Context.class);
+
+        Metacard parent = mock(Metacard.class);
+        List<Metacard> children = mock(List.class);
+
+        updateParent.handle(context, parent, children);
+
+        verify(updateField).updateField(parent, children, context);
+
+    }
+
+    /**
+     * Test that #end calls the catalog framework
+     */
+    @Test
+    public void testEnd() throws SourceUnavailableException, IngestException {
+
+        UpdateParent.UpdateField updateField = mock(UpdateParent.UpdateField.class);
+
+        UpdateParent updateParent = new UpdateParent(updateField);
+
+        UdpStreamProcessor udpStreamProcessor = mock(UdpStreamProcessor.class);
+        CatalogFramework catalogFramework = mock(CatalogFramework.class);
+        when(udpStreamProcessor.getCatalogFramework()).thenReturn(catalogFramework);
+        when(udpStreamProcessor.getMetacardUpdateInitialDelay()).thenReturn(1L);
+
+        Context context = mock(Context.class);
+        when(context.getUdpStreamProcessor()).thenReturn(udpStreamProcessor);
+
+        Metacard parent = mock(Metacard.class);
+
+        UpdateResponse updateResponse = mock(UpdateResponse.class);
+        Update update = mock(Update.class);
+        when(update.getNewMetacard()).thenReturn(parent);
+        when(updateResponse.getUpdatedMetacards()).thenReturn(Collections.singletonList(update));
+
+        when(catalogFramework.update(any(UpdateRequest.class))).thenReturn(updateResponse);
+
+        updateParent.end(context, parent);
+
+        verify(updateField).end(parent, context);
+
+        ArgumentCaptor<UpdateRequest> captor = ArgumentCaptor.forClass(UpdateRequest.class);
+
+        verify(catalogFramework).update(captor.capture());
+
+        assertThat(captor.getValue().getUpdates().get(0).getValue(), is(parent));
+
+    }
+
```

❗️ We should test that calling `end` before `update` throws an exception.
❗️ We should also test that the `RolloverActionException` is silently ignored in the `update()` method.